src/index.js

```diff
@@ -1,28 +1,71 @@
-import React from 'react';
+import React, { Component } from 'react';
 import ReactDOM from 'react-dom';
+import createHistory from 'history/createBrowserHistory'
 import {
   BrowserRouter as Router,
-  Route
+  Route,
+  matchPath
 } from 'react-router-dom'
 
-import App from './App';
+import Game from './components/Game';
 import Landing from './components/Landing';
 import './index.css';
 
-ReactDOM.render(
-  <Router>
-    <div className='app'>
-      <h1>Album Hangman</h1>
-      <div className='container'>
-        <Route exact path={'/'} component={Landing} />
-        <Route path={'/game/:username/:period?/:hideArtwork?'} component={App} />
+const GAME_PATH = '/game/:username?/:period?/:hideArtwork?';
+
+class App extends Component{
+  constructor(props){
+    super(props);
+
+    // Match current path to keep username and period in state
+    const history = createHistory();
+    this.match = matchPath(history.location.pathname, { path: GAME_PATH });
+    this.state = {
+      username: this.match ? this.match.params.username : '',
+      period: this.match ? this.match.params.period : '',
+      hideArtwork: this.match ? this.match.params.hideArtwork === 'hard' : false
+    }
+  }
+
+  handleSettingUpated = (name, value) => this.setState({ [name]: value });
```

Comment:
`handleSettingUpated` -> `handleSettingChange`
Also you can omit the value if you add a handler for the checkbox, and use `setState({ name })`.

Also maybe change `name` to `setting`?